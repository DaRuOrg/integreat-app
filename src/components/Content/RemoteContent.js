import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import style from './RemoteContent.css'

class RemoteContent extends React.Component {
  static propTypes = {
    dangerousHtmlContent: PropTypes.string.isRequired
  }

  render () {
    return <div className={cx(style.remoteContent, this.props.className)}
                dangerouslySetInnerHTML={{__html: (this.props.dangerousHtmlContent)}} />
  }
}

export default RemoteContent
