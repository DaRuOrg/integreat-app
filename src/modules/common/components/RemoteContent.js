import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import style from './RemoteContent.css'

class RemoteContent extends React.Component {
  static propTypes = {
    dangerouslySetInnerHTML: PropTypes.shape(
      PropTypes.shape({
        __html: PropTypes.string.isRequired
      }).isRequired
    ).isRequired,
    centered: PropTypes.bool
  }

  render () {
    const className = this.props.centered ? cx(style.remoteContent, style.centered) : style.remoteContent
    return <div className={className} dangerouslySetInnerHTML={this.props.dangerouslySetInnerHTML} />
  }
}

export default RemoteContent
