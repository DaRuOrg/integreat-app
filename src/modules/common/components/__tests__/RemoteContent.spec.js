import React from 'react'
import { shallow } from 'enzyme'
import RemoteContent from '../RemoteContent'

describe('RemoteContent', () => {
  test('should render', () => {
    expect(shallow(<RemoteContent dangerouslySetInnerHTML={{__html: '<div> Test html </div>'}} />)).toMatchSnapshot()
  })

  test('should render and have centered props', () => {
    expect(shallow(
      <RemoteContent isCentered dangerouslySetInnerHTML={{__html: '<div> Test html </div>'}} />
    )).toMatchSnapshot()
  })
})
