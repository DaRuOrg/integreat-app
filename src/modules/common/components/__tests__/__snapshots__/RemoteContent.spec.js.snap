// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteContent should render and be centered 1`] = `
<div
  className="remoteContent centered"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div> Test html </div>",
    }
  }
/>
`;

exports[`RemoteContent should render and not be centered 1`] = `
<div
  className="remoteContent"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div> Test html </div>",
    }
  }
/>
`;
